packages/extension/src/background/transactions/tracking.ts

```diff
@@ -28,7 +29,9 @@ export const transactionTracker: TransactionTracker = {
       // is smart enough to filter for just the pending transactions, as the rest needs no update
       allTransactions,
     )
-    await transactionsStore.push(updatedTransactions)
+    const [toBeRemoved, toBeKept] = partition(updatedTransactions, (tx) => tx.status === "REMOVED_FROM_MEMPOOL")
+    await transactionsStore.remove((tx) => toBeRemoved.some((toBeRemovedTx) => tx.hash === toBeRemovedTx.hash))
```

Comment:
Let's do it only when `toBeRemoved` is nonEmpty, just in case the remove action is expensive.